### ISSUE TITLE
Android App uploads the reports automatically even if 'Auto-upload only on Wi-Fi' is not checked.

### ISSUE THREAD

**Issue by CuriousLearner** [mozilla/MozStumbler#1698]
I've noticed this several times that the Android App uploads the reports automatically even if 'Auto-upload only on Wi-Fi' is not checked.

Here are the environment specifications and other details which might help:

1) Android Phone:  MotoG 2nd Gen 
2) Android version: 5.0.2
3) Kernel version: 3.4.42-g48d3b85
4) Build Number: LXB22.46-28
4) Mozilla Stumbler version: 1.7.12 d7b3156

Please let me know if any other details are required.

**Comment by cascheberg** [comment]
This is expected, but the wording might not be quite clear. The option could also be titled "Do not auto-upload on mobile networks", as this option only allows to reduce data usage by disallowing automatic uploads on mobile networks. So if you disable 'Auto-upload only on Wi-Fi', then the app will additionally automatically upload on mobile connections, as well as Wi-Fi still.
There is an issue filed for allowing to disable all automatic uploads, which might be what you are expecting, see #1691

**Comment by CuriousLearner** [comment]
Yes, all uploads should be disabled as the option mentioned is not clear. Thanks for clarification.

**Comment by crankycoder** [comment]
@CuriousLearner "all uploads should be disabled" isn't terribly helpful.  Many people upload using cell data plans.

Maybe we should break this up into 2 check boxes.

1) Allow Upload on Wifi
2) Allow Upload on Cell

Default to having #1 checked and require that at least one is checked off.
